File: nhglib-core/src/io/github/movementspeed/nhglib/runtime/fsm/interfaces/EngineStateListener.java
package io.github.movementspeed.nhglib.runtime.fsm.interfaces;

/**
 * Created by Fausto Napoli on 02/11/2016.
 */
public interface EngineStateListener {
    void onStart();

    void onInitialized();

    void onUpdate(float delta);

    void onPause();

    void onClose();

    void onResize(int width, int height);

    void onDispose();
}

File: nhglib-core/src/io/github/movementspeed/nhglib/core/fsm/interfaces/EngineStateListener.kt
package io.github.movementspeed.nhglib.core.fsm.interfaces

/**
 * Created by Fausto Napoli on 02/11/2016.
 */
interface EngineStateListener {
    fun onStart()

    fun onInitialized()

    fun onUpdate(delta: Float)

    fun onPause()

    fun onClose()

    fun onResize(width: Int, height: Int)

    fun onDispose()
}
